packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js

```diff
@@ -0,0 +1,285 @@
+import {LightningElement, api, track} from 'lwc';
+import caseClassificationTitle from '@salesforce/label/c.quantic_CaseClassificationTitle';
+import moreTopics from '@salesforce/label/c.quantic_MoreTopics';
+import selectOption from '@salesforce/label/c.quantic_SelectOption';
+import {
+  registerComponentForInit,
+  initializeWithHeadless,
+} from 'c/quanticHeadlessLoader';
+
+/** @typedef {import("coveo").CaseAssistEngine} CaseAssistEngine */
+/** @typedef {import("coveo").CaseField} CaseField */
+
+/**
+ * The `QuanticCaseClassification` component is a section for a user to classify his case aided by suggestions provided by Coveo Case Assist. There is also a dropdown available to see all available values for a given category.
+ *
+ * @category Case Assist
+ * @example
+ * <c-quantic-case-classification engine-id={engineId} field-name="sfpriority" options={options} required label="Which topic is related to your issue?" select-placeholder="More topics" max-choices="4" message-when-value-missing="Select an option"></c-quantic-case-classification>
+ */
+export default class QuanticCaseClassification extends LightningElement {
+  labels = {
+    caseClassificationTitle,
+    moreTopics,
+    selectOption,
+  };
+
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+
+  /**
+   * The field of the case to be classified.
+   */
+  @api fieldName;
+
+  /**
+   * All the possible values of a given category.
+   * @api
+   * @type {Array<object>}
+   * @defaultValue `[]`
+   */
+  @api options = [];
+
+  /**
+   * Tells whether the input is required or not.
+   * @api
+   * @type {boolean}
+   * @defaultValue `false`
+   */
+  @api required = false;
+
+  /**
+   * The label to be shown to the user.
+   * @api
+   * @type {string}
+   * @defaultValue `'Which topic relates to your issue?'`
+   */
+  @api label = this.labels.caseClassificationTitle;
+
+  /**
+   * The placeholder of the combo box input.
+   * @api
+   * @type {string}
+   * @defaultValue `'More Topics'`
+   */
+  @api selectPlaceholder = this.labels.moreTopics;
+
+  /**
+   * The maximum number of choices to be displayed, a choice can be a suggestion, an inline option or the select dropdown.
```

Comment:
I would like to understand better this option too.
According to the tests, I think `1, 0 and -1` all have the same effect basically?

Reply:
Exactly, in other words the minimum value  that `maxChoices` can take is 1.  The logic behind this is that max choices represents the maximum number of choices we propose to the user, and a single choice can be a suggestion, an inline option or the select drop down. so for example if maxChoices is set to 3 we can show 2 suggestions and the select drop down. if maxChoices is set to 1 we only can show the select drop down. Having maxChoices set to 0 means that we don't want anything no suggestion and no inline option and no select drop down and this will make our component kind of useless. so maxChoices is always greater or equal to 1. and in the case where the user gives maxChoices a negative value or 0, we always reset the value to 1.